src/nerf/gun.gleam

```diff
@@ -1,47 +1,37 @@
 //// Low level bindings to the gun API. You typically do not need to use this.
 //// Prefer the other modules in this library.
 
-import gleam/http.{Header}
-import gleam/erlang/charlist.{Charlist}
-import gleam/dynamic.{Dynamic}
-import gleam/string_builder.{StringBuilder}
-import gleam/bit_builder.{BitBuilder}
+import gleam/http.{type Header}
+import gleam/erlang/charlist.{type Charlist}
+import gleam/dynamic.{type Dynamic}
+import gleam/string_builder.{type StringBuilder}
+import gleam/bit_builder.{type BitBuilder}
 
-pub external type StreamReference
+pub opaque type StreamReference
 
-pub external type ConnectionPid
+pub opaque type ConnectionPid
 
 pub fn open(host: String, port: Int) -> Result(ConnectionPid, Dynamic) {
   open_erl(charlist.from_string(host), port)
 }
 
-pub external fn open_erl(Charlist, Int) -> Result(ConnectionPid, Dynamic) =
-  "gun" "open"
+@external(erlang, "gun", "open")
+pub fn open_erl(host: Charlist, port: Int) -> Result(ConnectionPid, Dynamic)
 
-pub external fn await_up(ConnectionPid) -> Result(Dynamic, Dynamic) =
-  "gun" "await_up"
+@external(erlang, "gun", "await_up")
+pub fn await_up(conn: ConnectionPid) -> Result(Dynamic, Dynamic)
 
-pub external fn ws_upgrade(
-  ConnectionPid,
-  String,
-  List(Header),
-) -> StreamReference =
-  "gun" "ws_upgrade"
+@external(erlang, "gun", "ws_upgrade")
+pub fn ws_upgrade(
+  conn: ConnectionPid,
+  path: String,
+  headers: List(Header),
+) -> StreamReference
 
 pub type Frame {
   Close
   Text(String)
-  Binary(BitString)
+  Binary(BitArray)
   TextBuilder(StringBuilder)
   BinaryBuilder(BitBuilder)
 }
-
-external type OkAtom
-
-external fn ws_send_erl(ConnectionPid, Frame) -> OkAtom =
-  "nerf_ffi" "ws_send_erl"
-
-pub fn ws_send(pid: ConnectionPid, frame: Frame) -> Nil {
```

Comment:
I think this function has been mistakenly removed?

Reply:
individual commits had a bit more context.

`ws_send/2` and `ws_send_erl` were moved in in src/nerf/websocket.gleam since they require Connection (not just ConnectionPid). 

Now another approach would be to split the types into their own file, so we can import them everywhere, or … circular imports (are they allowed?)

Reply:
Please revert this breaking change, thank you